### PR TITLE
#315: Fixed unicode surrogates bug with Python 3.12.3 and 3.11.9

### LINKED ISSUE #315: Unicode issue when sending with python 3.11.9, 3.12.3
It looks like a change added in python 3.11.9 [#76511 Email parser creates a message object that can't be flattened](https://github.com/python/cpython/issues/76511) has cuased a bug in django-ses.
If you send a message with unicode in the body you get the following error

```
File "/usr/local/lib/python3.11/site-packages/django/core/mail/message.py", line 169, in <genexpr>
    len(line.encode()) > RFC5322_EMAIL_LINE_LENGTH_LIMIT
        ^^^^^^^^^^^^^
UnicodeEncodeError: 'utf-8' codec can't encode characters in position 3-5: surrogates not allowed
```

You can replicate this by updating the test_send_mail https://github.com/django-ses/django-ses/blob/83b32eca995b7eb495395d860a61a44533e19e5a/tests/test_backend.py#L115-L127

to
```diff
    def test_send_mail(self):
        settings.AWS_SES_CONFIGURATION_SET = None

        unicode_from_addr = 'Unicode Name óóóóóó <from@example.com>'

-       send_mail('subject', 'body', unicode_from_addr, ['to@example.com'])
+       send_mail('subject', 'body with unicode óóóóóó', unicode_from_addr, ['to@example.com'])
        message = self.outbox.pop()
        mail = email.message_from_string(smart_str(message['RawMessage']['Data']))
        self.assertTrue('X-SES-CONFIGURAITON-SET' not in mail.keys())
        self.assertEqual(mail['subject'], 'subject')
        self.assertEqual(mail['from'], self._rfc2047_helper(unicode_from_addr))
        self.assertEqual(mail['to'], 'to@example.com')
-       self.assertEqual(mail.get_payload(), 'body')
+       self.assertEqual(mail.get_payload(), 'body with unicode óóóóóó')
```

This test will pass under python 3.11.8 but fail under 3.11.9

### PR DESCRIPTION
**Pull request by GitRon**
fixed #315

### CODE REVIEW

**Review by pcraciunoiu — APPROVED** [review]
Tested locally as well on 3.11.9 and 3.12.3, LGTM

Thank you!

File: tests/test_backend.py [hunk 1]
```diff
@@ -115,17 +115,31 @@ def test_rfc2047_helper(self):
...
 
+    def test_send_mail_unicode_body(self):
+        settings.AWS_SES_CONFIGURATION_SET = None
+
+        unicode_from_addr = 'Unicode Name óóóóóó <from@example.com>'
+
+        send_mail('Scandinavian', 'Sören & Björn', unicode_from_addr, ['to@example.com'])
+        message = self.outbox.pop()
+        mail = email.message_from_string(smart_str(message['RawMessage']['Data']))
+        self.assertTrue('X-SES-CONFIGURAITON-SET' not in mail.keys())
```
Comment (ryanhiebert) [inline tests/test_backend.py:137]: It looks like this was spelled incorrectly?

Comment (pcraciunoiu) [reply tests/test_backend.py:137]: Good catch--you're not the first, it was fixed in this PR: #319

Comment (ryanhiebert) [reply tests/test_backend.py:137]: ok good! sorry for the extra noise!

**Pull request merged**